islandora_compound_object.module

```diff
@@ -450,18 +470,15 @@ EOQ;
     uasort($results, $sort);
 
     foreach ($results as $result) {
-      $object = islandora_object_load($result['object']['value']);
-      if (islandora_object_access(ISLANDORA_VIEW_OBJECTS, $object)) {
```

Comment:
I kind of see a compound object a lot like a collection where we have many children and can query for their availability using solr or sparql. Is there a way to satisfy the access restrictions you are thinking about without loading every object and then calling islandora_object_access on it?